.dockerignore

```diff
@@ -0,0 +1,44 @@
+# Ignore all files that should:
+# a) not be in a release tarball
+# b) all other unncessary files that are not required to run Zulip
+# (examples REAME, Vagrant stuff, docker-compose file, etc.)
+#
+*.pyc
+*~
+/prod-static
+*.sw[po]
+*.DS_Store
+.kdev4
+.idea
+zulip.kdev4
+coverage/
+.coverage
+/queue_error
+.kateproject.d/
+.kateproject
+*.kate-swp
+*.sublime-project
+*.sublime-workspace
+.vagrant
+/zproject/dev-secrets.conf
+static/js/bundle.js
+static/generated/emoji
+static/generated/github-contributors.json
+static/locale/language_options.json
```

Comment:
Several of these files are needed in production, but can be generated; can you add a comment explaining these?

It might make sense to use the approach we use in `tools/build-release-tarball`, where we start by basically using `git archive` to create a fresh clone-type tarball/directory from the current git commit.  That would eliminate the need for this file to duplicate `.gitignore`.